### PR TITLE
Implement connector instance removal functionality and enhance UI int…

### PR DESCRIPTION
**Pull request by Kushagra1122**
…eractions

- Added `removeConnectorInstanceCaches` method to clear cached data for deleted connectors.
- Updated `InstanceCard` to prevent interaction with connectors that are being deleted.
- Enhanced `InstanceManagementPanel` to include a confirmation dialog for connector removal, with appropriate loading states and success/error notifications.
- Modified `SettingsTab` to display a danger zone for connector removal, including disabled states and tooltips for user guidance.
- Updated sync handling in `PersonalConnectorsPageContent` and `TeamConnectorsPageContent` to prevent actions on connectors marked as 'DELETING'.

## Description

[Provide a brief description of the changes in this PR]

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Is there an addition/change in API Request, Response? If yes, documentation is required)
- [ ] Performance improvement
- [ ] Code refactoring
- [ ] Security fix

## Related Issues

- Fixes #[issue number]
- Closes #[issue number]
- Related to #[issue number]

## How Has This Been Tested?

[Describe the tests you ran to verify your changes. Provide instructions so we can reproduce.]

### Test Configuration

- [ ] Unit tests pass
- [ ] Integration tests pass
- [ ] Manual testing completed
- [ ] Cross-browser testing (if applicable)
- [ ] Mobile responsiveness tested (if applicable)

## Core Functionality Testing

Please confirm that the following core functionalities are working as expected:

- [ ] **Search capabilities** are working correctly
- [ ] **Knowledge search** is functioning properly
- [ ] **Connector indexing** is working as expected
- [ ] **Citations** are displaying and linking correctly
- [ ] **Documentation** is updated at https://docs.pipeshub.com/introduction

## What You Have Tested

Please describe what you have specifically tested:

- [ ] Feature works in development environment
- [ ] Feature works in staging environment
- [ ] Error handling scenarios tested
- [ ] Edge cases considered and tested
- [ ] Performance impact assessed
- [ ] Security implications reviewed

### Test Results

[Provide details of your test results here]

## Screenshots/Videos

**Before:**
[Attach screenshots/videos showing the current behavior]

**After:**
[Attach screenshots/videos showing the new behavior]

**Note:** Screenshots are required before merge for UI changes.

## Code Quality Checklist

- [ ] My code follows the project's coding standards
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Documentation

- [ ] Documentation has been updated (if applicable)
- [ ] API documentation updated (if applicable)
- [ ] README updated (if applicable)
- [ ] Changelog updated (if applicable)

## Security Considerations

- [ ] No sensitive data is exposed
- [ ] Input validation is implemented where necessary
- [ ] Authentication/authorization is properly handled
- [ ] No security vulnerabilities introduced

## Breaking Changes

- [ ] This PR introduces breaking changes
- [ ] Migration guide provided (if applicable)
- [ ] Version bump required

If breaking changes are introduced, please describe them here:

[Describe any breaking changes and how users should adapt]

## Performance Impact

- [ ] No performance impact
- [ ] Performance improved
- [ ] Performance impact assessed and acceptable

[Describe any performance implications]

## Dependencies

- [ ] No new dependencies added
- [ ] New dependencies are necessary and approved
- [ ] Dependencies updated and tested

List any new dependencies:
- [Dependency name and version]

## Deployment Notes

[Any special deployment considerations or steps]

## Checklist Before Merge

- [ ] All tests are passing
- [ ] Code review completed and approved
- [ ] Documentation updated and reviewed
- [ ] Screenshots/videos attached for UI changes
- [ ] Core functionality verified
- [ ] Security review completed (if applicable)
- [ ] Performance impact assessed
- [ ] Ready for production deployment

## Additional Notes

[Any additional information that would be helpful for reviewers]

---

**For Reviewers:**

Please ensure all checklist items are completed before approving this PR. Pay special attention to:
- Core functionality testing
- Security implications
- Performance impact
- Documentation completeness

### CODE REVIEW

File: frontend-new/app/(main)/workspace/connectors/components/instance-panel/index.tsx [hunk 1]
```diff
@@ -44,6 +57,56 @@ export function InstanceManagementPanel() {
...
+      const raw = await ConnectorsApi.deleteConnectorInstance(id);
+      const data = raw as Record<string, unknown> | null | undefined;
+      if (
+        data &&
+        typeof data === 'object' &&
+        typeof data._key === 'string' &&
+        typeof data.type === 'string'
+      ) {
+        upsertConnectorInstance(data as Connector);
+      }
```
[the remark this answers is not in the document]
Comment (Kushagra1122) [reply frontend-new/app/(main)/workspace/connectors/components/instance-panel/index.tsx]: ll update the handler so we don’t treat the DELETE body as a full Connector replacement

File: frontend-new/app/(main)/workspace/connectors/components/instance-panel/index.tsx [hunk 2]
```diff
@@ -135,62 +213,119 @@ export function InstanceManagementPanel() {
...
-            onClick={handleManageConfiguration}
-            style={{ cursor: 'pointer' }}
-          >
-            <MaterialIcon name="settings" size={16} color="var(--gray-11)" />
-            Manage Configuration
-          </Button>
-        </Flex>
-      </Flex>
-    </WorkspaceRightPanel>
+          <AlertDialog.Content container={nestedModalHost} style={{ maxWidth: 440 }}>
```
[the remark this answers is not in the document]
Comment (Kushagra1122) [reply frontend-new/app/(main)/workspace/connectors/components/instance-panel/index.tsx:314]: We’re on @radix-ui/themes, where AlertDialog.Content is typed to accept container and the implementation forwards it to AlertDialog

**Comment by rish664** [comment]
@claude review the pr

**Review by rish664 — DISMISSED** [review]

File: frontend-new/app/(main)/workspace/connectors/api.ts [hunk 3]
```diff
@@ -12,6 +12,48 @@ import type {
...
+/** Normalized DELETE /connectors/:id body for optimistic UI merge. */
+export type DeleteConnectorInstanceMerge = {
+  _key: string;
+  type: string;
+  status: string | null;
+};
+
+function resolveDeleteInstanceResponseStatus(body: Record<string, unknown>): string | null {
+  if (!('status' in body)) {
+    return 'DELETING';
```
Comment (rish664) [inline frontend-new/app/(main)/workspace/connectors/api.ts]: save it in constants.
`DELETING`

File: frontend-new/app/(main)/workspace/connectors/api.ts [hunk 4]
```diff
@@ -12,6 +12,48 @@ import type {
...
+    return s;
+  }
+  if (s === null) {
+    return 'DELETING';
+  }
+  return 'DELETING';
+}
+
+function parseDeleteConnectorInstanceBody(
+  data: unknown,
```
Comment (rish664) [inline frontend-new/app/(main)/workspace/connectors/api.ts]: this is not correct 
`data: unknown,`

File: frontend-new/app/(main)/workspace/connectors/api.ts [hunk 5]
```diff
@@ -12,6 +12,48 @@ import type {
...
+  fallbackConnectorId: string
+): DeleteConnectorInstanceMerge {
+  if (data && typeof data === 'object') {
+    const o = data as Record<string, unknown>;
```
Comment (rish664) [inline frontend-new/app/(main)/workspace/connectors/api.ts]: unknown - do not use it

File: frontend-new/app/(main)/workspace/connectors/personal/page.tsx [hunk 6]
```diff
@@ -314,7 +314,7 @@ function PersonalConnectorsPageContent() {
 
   const handleStartSync = useCallback(
     async (instance: ConnectorInstance) => {
-      if (!instance._key) return;
+      if (!instance._key || instance.status === 'DELETING') return;
```
Comment (rish664) [inline frontend-new/app/(main)/workspace/connectors/personal/page.tsx]: use constants

File: frontend-new/app/(main)/workspace/connectors/api.ts (hunk 3 above)
Comment (Kushagra1122) [reply frontend-new/app/(main)/workspace/connectors/api.ts]: done

File: frontend-new/app/(main)/workspace/connectors/api.ts (hunk 4 above)
Comment (Kushagra1122) [reply frontend-new/app/(main)/workspace/connectors/api.ts]: fixed

File: frontend-new/app/(main)/workspace/connectors/api.ts (hunk 5 above)
Comment (Kushagra1122) [reply frontend-new/app/(main)/workspace/connectors/api.ts]: fixed

File: frontend-new/app/(main)/workspace/connectors/personal/page.tsx (hunk 6 above)
Comment (Kushagra1122) [reply frontend-new/app/(main)/workspace/connectors/personal/page.tsx]: fixed

File: frontend-new/app/(main)/workspace/connectors/api.ts [hunk 7]
```diff
@@ -9,9 +9,58 @@ import type {
...
+    return CONNECTOR_INSTANCE_STATUS.DELETING;
+  }
+  return CONNECTOR_INSTANCE_STATUS.DELETING;
+}
+
+function parseDeleteConnectorInstanceBody(
+  data: DeleteConnectorInstanceResponseBody | null | undefined,
+  fallbackConnectorId: string
+): DeleteConnectorInstanceMerge {
+  if (data && typeof data === 'object' && typeof data._key === 'string') {
```
Comment (rish664) [inline frontend-new/app/(main)/workspace/connectors/api.ts:50]: @Kushagra1122 what is this condition?

Comment (Kushagra1122) [reply frontend-new/app/(main)/workspace/connectors/api.ts:50]: It means “we got a real JSON body from the DELETE response with a string _key.” If data is missing, not an object, or _key isn’t a string (empty body, wrong shape, etc.), we skip parsing and use fallbackConnectorId plus the default “deleting” status instead.

Comment (rish664) [reply frontend-new/app/(main)/workspace/connectors/api.ts:50]: > It means “we got a real JSON body from the DELETE response with a string _key.” If data is missing, not an object, or _key isn’t a string (empty body, wrong shape, etc.), we skip parsing and use fallbackConnectorId plus the default “deleting” status instead.

if data is of type DeleteConnectorInstaceRespmseBody i do not understand why these checks needed?

Comment (Kushagra1122) [reply frontend-new/app/(main)/workspace/connectors/api.ts:50]: > > It means “we got a real JSON body from the DELETE response with a string _key.” If data is missing, not an object, or _key isn’t a string (empty body, wrong shape, etc.), we skip parsing and use fallbackConnectorId plus the default “deleting” status instead.
> 
> if data is of type DeleteConnectorInstaceRespmseBody i do not understand why these checks needed?

TypeScript only applies at compile time data still comes from the wire as parsed JSON, so at runtime it can be null/undefined, an empty object, or missing _key even though the type allows that (_key is optional). The checks are just defensive narrowing so we don’t optimistically merge on a bad body.

If we’re happy assuming the DELETE handler always returns a full body with _key, we can drop this guard and simplify—happy to remove it in a follow-up if you prefer.

**Review by rish664 — APPROVED** [review]
